File: NetBet/ApiControllers/EventController.fs
namespace NetBet.Controllers

open System
open System.Collections.Generic
open System.Linq
open Microsoft.AspNetCore.Mvc
open DbTypes
open DtoTypes

[<Route("api/[controller]/[action]")>]
[<ApiController>]
type EventController () =
    inherit ControllerBase()
    
    [<HttpGet>]
    member __.Get() =
        let players = SeasonService.getAllPlayers()
        ActionResult<Player[]>(players)

    [<HttpGet("{eventid}")>]
    member __.GetById(eventid: int) =
        let evt = EventService.getEventByID eventid
        ActionResult<Event>(evt)

    [<HttpGet("{seasonid}")>]
    member __.GetEventsForSeason(seasonid: int) =
        let evts = EventService.getEventsForSeason seasonid
        ActionResult<Event[]>(evts)

    [<HttpGet("{eventid}")>]
    member __.GetEventWithMatches(eventid: int) = 
        let evt = EventService.getEventAndMatches eventid
        ActionResult<EventWithMatches>(evt)

    [<HttpGet("{eventid}")>]
    member __.GetFullEvent(eventid: int) = 
        let evt = EventService.getFullEvent eventid
        ActionResult<FullEvent>(evt)
        
    [<HttpGet>]
    member __.GetEmpty() =
        let emptyevt = 
            { ID        = 0
              SeasonID  = 0
              Name      = ""
              StartTime = DateTime.Now }
        let emptyEventWithMatches = 
            { Event = emptyevt
              Matches = [| |] }
        ActionResult<EventWithMatches>(emptyEventWithMatches)
        
    [<HttpPost>]
    member __.CreateOrUpdate([<FromBody>] evt: EventWithMatches) =
        EventService.createEventWithMatches evt
        
    [<HttpDelete("{id}")>]
    member __.Delete(id:int) =
        EventService.deleteEvent id

File: NetBet.WebApp/ApiControllers/EventController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using static DbTypes;
using static DtoTypes;

namespace NetBet.WebApp.ApiControllers
{
    [Route("api/[controller]/[action]")]
    [ApiController]
    public class EventController : ControllerBase
    {   
        [HttpGet("{eventid}")]
        public Event GetById(int eventid)
        {
            return EventService.getEventByID(eventid);
        }

        [HttpGet("{seasonid}")]
        public Event[] GetEventsForSeason(int seasonid)
        {
            return EventService.getEventsForSeason(seasonid);
        }

        [HttpGet("{eventid}")]
        public EventWithPrettyMatches GetEventWithMatches(int eventid)
        {
            return EventService.getEventAndMatches(eventid);
        }

        [HttpGet("{eventid}")]
        public FullEvent GetFullEvent(int eventid)
        {
            return EventService.getFullEvent(eventid);
        }

        [HttpGet("{eventName}")]
        public int GetEventIDByName(string eventName)
        {
            return EventService.getEventIDByName(eventName);
        }

        [HttpGet]
        public EventWithPrettyMatches[] GetUpcomingEventsFromWeb()
        {
            return EventService.getUpcomingEventsFromWeb();
        }
        
        [HttpPost]
        public void CreateOrUpdate([FromBody] EventWithPrettyMatches evt)
        {
            EventService.createOrUpdateEventWithMatches(evt);
        }
        
        [HttpDelete("{id}")]
        public void Delete(int id)
        {
            EventService.deleteEvent(id);
        }
    }
}
